### ISSUE TITLE
Changes are breaking something commit 1f4056d

### ISSUE THREAD

**Issue by Karmel0x** [smalot/pdfparser#359]
Unfortunately I can't provide sample PDF this time.
Output is correct before commit 1f4056d.

Output with latest commit / release:
```
Notice: Undefined offset: 67 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 117 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 114 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 114 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 105 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 99 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 117 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 108 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 117 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 109 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 32 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 86 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 105 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 116 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 97 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 101 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 32 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 48 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 55 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 46 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 48 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091
Notice: Undefined offset: 53 in pdfparser\src\Smalot\PdfParser\Encoding\PostScriptGlyphs.php on line 1091

... more than 5000 similary errors here
```

text:

ż ś ą ą żą ń ą ą ą ą ą ą ą ą ę ś ę ń ą ą ą ń ę ż ą ś ś ż ą ą ż ą ą ą ż ą ę ą ń ż ń ą ą ś ą ą ą ę ż ęż ń ę ę ą ą ą Ś Ś ą ę ż żą ż ż ż ą ś ąś ń ń ń ń ą ę ś ść Ś ść ść ść ść ś ść ę ść ą ść ść ść ść ę ś ż ę ą ą ą

**Comment by k00ni** [comment]
It seems that there are glyphs in use which are not part of `PostScriptGlyphs.php`. Because this functionality is from @Connum, lets see what he has to say.

**Comment by Connum** [comment]
It would be really great to have a sample, otherwise I'm not sure how to debug this properly. Maybe you can remove any sensitive content and just leave in a few sentences or paragraphs that cause the issue?

**Comment by Connum** [comment]
Also, could you please try whether the issue happens with this branch as well? https://github.com/skyfms/pdfparser/tree/fix_encoding
Because I didn't implement the functionality originally, but rather adapted it to integrate it into the current state of the library. That way I could see if I introduced the breaking change myself or of the original code was already faulty.

**Comment by Karmel0x** [comment]
Thought after editing error will disappear but it's still there, I just deleted some pages, so I can provide PDF now: [cv_Zj.pdf](https://github.com/smalot/pdfparser/files/5429471/cv_Zj.pdf)

Problem occurs in this branch too.

**Comment by k00ni** [comment]
Can we add the PDF file (https://github.com/smalot/pdfparser/files/5429471/cv_Zj.pdf) to our test environment? Is it free of charge and has no obligations?

**Comment by Karmel0x** [comment]
I am not the author but I've removed sensitive informations, so.. probably.

**Comment by k00ni** [comment]
@Connum: #360 is another issue which reports a problem with `PostScriptGlyphs`, which was added lately. What needs to be done to fix the underlying problem?

**Comment by Connum** [comment]
> @Connum: #360 is another issue which reports a problem with `PostScriptGlyphs`, which was added lately. What needs to be done to fix the underlying problem?

Sorry for not responding, I'm quite busy at the moment with several work projects in parallel, but I haven't forgotten about the issue and will look at it as soon as I find the time!

**Comment by Connum** [comment]
Finally got around looking into this. I just created PR #362 to handle this issue.
The PDF content appears to be some kind of legal text or at least a factual report and also seems to be a small enough excerpt, so I don't see why we couldn't use it for a test case (which I did).

Please note the following, which I also added as a comment in the related test case:
```
Note that some characters are missing from the parsed text or wrongly decoded,
like the "ł" in przepływu or "ó" in osób. This was already the case before the
PR that caused this issue and is not currently covered by this test case.
However, these issues should be addressed in the future and their fixes can
then be incorporated into this test by uncommenting the two assertions below.
```

**Comment by k00ni** [comment]
@Karmel0x can you test again with the patch @Connum provided? Do you approve the fix?

**Comment by Karmel0x** [comment]
If it's reading above pdf then it's fine and it looks like it is, original one too.
